src/Model.jl

```diff
@@ -21,8 +21,14 @@ struct Model{T <: Real, VT <: Real}
     unit_cell_volume::T
     recip_cell_volume::T
 
-    # Electrons, occupation and smearing function
-    n_electrons::Int  # usually consistent with `atoms` field, but doesn't have to
+    # If not `nothing` then do computations in the canonical ensemble, where
```

Comment:
this is not necessarily the canonical ensemble (as the temperature might be zero). I'm less and less convinced this vocabulary of ensembles is adequate here...